Compilation failure with GSL 2.1

Hello,

I get the following error on compilation : 
<code>
gfortran -DHAVE_CONFIG_H -I.  -I. -I.   -g -O2 -c -o fgsl.o fgsl.F90
api/spmatrix.finc:75.36:
    Included at fgsl.F90:1939:

  call gsl_spmatrix_cumsum(n, c_loc(c))
                                    1
Error: Assumed-shape array 'c' at (1) cannot be an argument to the procedure 'c_loc' because it is not C interoperable
api/fit.finc:47.39:
    Included at fgsl.F90:1934:

    fgsl_fit_wmul = gsl_fit_wmul(c_loc(x), xstride, c_loc(w), wstride, &
                                       1
Error: Assumed-shape array 'x' at (1) cannot be an argument to the procedure 'c_loc' because it is not C interoperable
api/fit.finc:38.37:
    Included at fgsl.F90:1934:

    fgsl_fit_mul = gsl_fit_mul(c_loc(x), xstride, c_loc(y), ystride, n, c1, &
                                     1
Error: Assumed-shape array 'x' at (1) cannot be an argument to the procedure 'c_loc' because it is not C interoperable
api/fit.finc:23.45:
    Included at fgsl.F90:1934:

    fgsl_fit_wlinear = gsl_fit_wlinear(c_loc(x), xstride, c_loc(w), wstride, &
                                             1
Error: Assumed-shape array 'x' at (1) cannot be an argument to the procedure 'c_loc' because it is not C interoperable
api/fit.finc:14.43:
    Included at fgsl.F90:1934:

    fgsl_fit_linear = gsl_fit_linear(c_loc(x), xstride, c_loc(y), ystride, n, c
                                           1
Error: Assumed-shape array 'x' at (1) cannot be an argument to the procedure 'c_loc' because it is not C interoperable
api/dht.finc:34.52:
    Included at fgsl.F90:1929:

    fgsl_dht_apply = gsl_dht_apply(t%gsl_dht, c_loc(f_in), c_loc(f_out))
                                                    1
Error: Assumed-shape array 'f_in' at (1) cannot be an argument to the procedure 'c_loc' because it is not C interoperable
api/wavelet.finc:156.64:
    Included at fgsl.F90:1928:

         gsl_wavelet2d_nstransform_inverse(w%gsl_wavelet, c_loc(data), &
                                                                1
Error: Assumed-shape array 'data' at (1) cannot be an argument to the procedure 'c_loc' because it is not C interoperable
api/wavelet.finc:146.64:
    Included at fgsl.F90:1928:

         gsl_wavelet2d_nstransform_forward(w%gsl_wavelet, c_loc(data), &
                                                                1
Error: Assumed-shape array 'data' at (1) cannot be an argument to the procedure 'c_loc' because it is not C interoperable
api/wavelet.finc:136.80:
    Included at fgsl.F90:1928:

gsl_wavelet2d_nstransform = gsl_wavelet2d_nstransform(w%gsl_wavelet, c_loc(data
                                                                           1
Error: Assumed-shape array 'data' at (1) cannot be an argument to the procedure 'c_loc' because it is not C interoperable
api/wavelet.finc:98.62:
    Included at fgsl.F90:1928:

         gsl_wavelet2d_transform_inverse(w%gsl_wavelet, c_loc(data), &
                                                              1
Error: Assumed-shape array 'data' at (1) cannot be an argument to the procedure 'c_loc' because it is not C interoperable
api/wavelet.finc:88.62:
    Included at fgsl.F90:1928:

         gsl_wavelet2d_transform_forward(w%gsl_wavelet, c_loc(data), &
                                                              1
Error: Assumed-shape array 'data' at (1) cannot be an argument to the procedure 'c_loc' because it is not C interoperable
api/wavelet.finc:78.76:
    Included at fgsl.F90:1928:

   fgsl_wavelet2d_transform = gsl_wavelet2d_transform(w%gsl_wavelet, c_loc(data
                                                                           1
Error: Assumed-shape array 'data' at (1) cannot be an argument to the procedure 'c_loc' because it is not C interoperable
api/wavelet.finc:68.88:
    Included at fgsl.F90:1928:

let_transform_inverse = gsl_wavelet_transform_inverse(w%gsl_wavelet, c_loc(data
                                                                           1
Error: Assumed-shape array 'data' at (1) cannot be an argument to the procedure 'c_loc' because it is not C interoperable
api/wavelet.finc:59.88:
    Included at fgsl.F90:1928:

let_transform_forward = gsl_wavelet_transform_forward(w%gsl_wavelet, c_loc(data
                                                                           1
Error: Assumed-shape array 'data' at (1) cannot be an argument to the procedure 'c_loc' because it is not C interoperable
api/wavelet.finc:50.72:
    Included at fgsl.F90:1928:

    fgsl_wavelet_transform = gsl_wavelet_transform(w%gsl_wavelet, c_loc(data), &
                                                                        1
Error: Assumed-shape array 'data' at (1) cannot be an argument to the procedure 'c_loc' because it is not C interoperable
api/ode.finc:547.25:
    Included at fgsl.F90:1924:

         t, t1, h, c_loc(y))
                         1
Error: Assumed-shape array 'y' at (1) cannot be an argument to the procedure 'c_loc' because it is not C interoperable
api/ode.finc:519.15:
    Included at fgsl.F90:1924:

         c_loc(y0), c_loc(yerr), c_loc(dydt), c_loc(h))
               1
Error: Assumed-shape array 'y0' at (1) cannot be an argument to the procedure 'c_loc' because it is not C interoperable
api/ode.finc:491.15:
    Included at fgsl.F90:1924:

         c_loc(scale_abs), size(scale_abs, kind=fgsl_size_t))
               1
Error: Assumed-shape array 'scale_abs' at (1) cannot be an argument to the procedure 'c_loc' because it is not C interoperable
api/ode.finc:464.10:
    Included at fgsl.F90:1924:

    c_loc(y), c_loc(yerr), &
          1
Error: Assumed-shape array 'y' at (1) cannot be an argument to the procedure 'c_loc' because it is not C interoperable
api/ode.finc:219.46:
    Included at fgsl.F90:1924:

         dydt%gsl_odeiv2_system, t, h0, c_loc(y))
                                              1
Error: Assumed-shape array 'y' at (1) cannot be an argument to the procedure 'c_loc' because it is not C interoperable
api/ode.finc:206.25:
    Included at fgsl.F90:1924:

         t, t1, h, c_loc(y))
                         1
Error: Assumed-shape array 'y' at (1) cannot be an argument to the procedure 'c_loc' because it is not C interoperable
api/ode.finc:163.56:
    Included at fgsl.F90:1924:

         c%gsl_odeiv2_control, s%gsl_odeiv2_step, c_loc(y), c_loc(yerr), c_loc(
                                                        1
Error: Assumed-shape array 'y' at (1) cannot be an argument to the procedure 'c_loc' because it is not C interoperable
api/ode.finc:128.15:
    Included at fgsl.F90:1924:

         c_loc(scale_abs), size(scale_abs, kind=fgsl_size_t))
               1
Error: Assumed-shape array 'scale_abs' at (1) cannot be an argument to the procedure 'c_loc' because it is not C interoperable
api/ode.finc:102.10:
    Included at fgsl.F90:1924:

    c_loc(y), c_loc(yerr), c_loc(dydt_in), c_loc(dydt_out), dydt%gsl_odeiv2_sys
          1
Error: Assumed-shape array 'y' at (1) cannot be an argument to the procedure 'c_loc' because it is not C interoperable
api/ntuple.finc:33.34:
    Included at fgsl.F90:1921:

            gsl_ntuple_open(c_loc(lname), data, size)
                                  1
Error: CHARACTER argument 'lname' to 'c_loc' at (1) must have a length of 1
Fatal Error: Error count reached limit of 25.
make: *** [fgsl.o] Erreur 1
</code>

I tried with both the latest stable release (1.1.0) and the git master, same error.

gsl is version 2.1, compiled from the sources.

Please let me know if any other information is useful...

--- a/configure.ac
+++ b/configure.ac
@@ -68,6 +68,29 @@ if test "x$IEEE_EXCEPTIONS" = xyes ; then
 	AC_DEFINE([IEEE_EXCEPTIONS],[],[Fortran 2003 ieee_exceptions intrinsic module found])
 fi
 
+#check for target support
+AC_FC_SRCEXT([F90],[
+	AC_MSG_CHECKING([whether the fortran compiler supports the 2008 feature: c_loc for targeted arrays])
+	AC_LANG_PUSH(Fortran)
+	AC_COMPILE_IFELSE([[
+SUBROUTINE test(c)
+USE, INTRINSIC :: iso_c_binding
+IMPLICIT NONE
+INTEGER(c_size_t), DIMENSION(:), INTENT(inout), TARGET:: c
+TYPE(C_PTR) :: ptr
+
+ptr = C_LOC(c)
+
+ENDSUBROUTINE
+	]], [C_LOC_TARGET=yes], [C_LOC_TARGET=no])
+	AC_LANG_POP(Fortran)
+	AC_MSG_RESULT([$C_LOC_TARGET])
+])
+
+if test "x$C_LOC_TARGET" = xno; then
+	AC_MSG_ERROR([Fortran compiler does not support a required feature from the 2008 standard.])
+fi
+
 case "$host" in
 	*mingw*)
 		FGSL_LDFLAGS="-no-undefined"

